compliance_checker/cf/cf.py

```diff
@@ -3698,6 +4035,286 @@ def check_cell_measures(self, ds):
+
+        if (not all([
+            x in var.ncattrs() for x in [
+                'reference_ellipsoid_name', 'prime_meridian_name',
+                'horizontal_datum_name'
+                ]
+            ])) and (any([
+                x in var.ncattrs() for x in [
+                    'reference_ellipsoid_name', 'prime_meridian_name',
+                    'horizontal_datum_name'
+                    ]
+                ])):
+
+            return (False, msg)
+
+        else:
+            return (True, msg)
+        
+
+    def _get_projdb_conn(self):
+        """
+        Return a SQLite Connection to the PROJ database.
+
+        Returns:
+            sqlite3.Connection
+        """
+
+        proj_db_path = os.path.join(pyproj.datadir.get_data_dir(), 'proj.db')
+        return sqlite3.connect(proj_db_path)
+
+    def _evaluate_geographic_crs_name(self, val):
+        """
+        Evalute the condition for the geographic_crs_name attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        conn = self._get_projdb_conn()
+        query_str = ('SELECT 1 FROM geodetic_crs WHERE name = ? '
+                     'UNION ALL ' # need union in case contained in other tables
+                     'SELECT 1 FROM alias_name WHERE alt_name = ? '
+                     'AND table_name = \'geodetic_crs\' LIMIT 1')
+       
+        # try to find the value in the database
+        res_set = conn.execute(query_str, (val, val))
+
+        # does it exist? if so, amt returned  be > 1
+        res = len(res_set.fetchall()) > 0
+
+        return (
+            res,
+            'geographic_crs_name must correspond to a valid OGC WKT GEOGCS name'
+        )
+
+    def _evaluate_geoid_name(self, val):
+        """
+        Evalute the condition for the geod_name attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        conn = self._get_projdb_conn()
+        query_str = ('SELECT 1 FROM vertical_datum WHERE name = ? ' 
+                     'UNION ALL '
+                     'SELECT 1 FROM alias_name WHERE alt_name = ? '
+                     'AND table_name = \'vertical_datum\' LIMIT 1')
+
+        # try to find the value in the database
+        res_set = conn.execute(query_str, (val, val))
+
+        # does it exist? if so, amt returned  be > 1
+        res = len(res_set.fetchall()) > 0
+
+        return (
+            res,
+            'geoid_name must correspond to a valid OGC WKT VERT_DATUM name'
+        )
+
+    def _evaluate_geopotential_datum_name(self, val):
+        """
+        Evalute the condition for the geogpotential_datum_name attribute.
+
+        :param val: value to be tested
+        :rtype tuple
+        :return two-tuple of (bool, str)
+        """
+
+        conn = self._get_projdb_conn()
+        query_str = ('SELECT 1 FROM vertical_datum WHERE name = ? ' 
```

Comment:
A lot of the format of these database methods is roughly the same.  You could make a helper function that takes a query string and an error message/test context and reduce the amount of repetition somewhat.

Reply:
See newest commit for changes